pages/memory.js

```diff
@@ -3,46 +3,62 @@ import {
     Container, Board, Card, CardFront,
     CardBack
 } from "../components/Memory";
+import { Head } from "../components/Header";
 
 const board = ["🤖", "👽", "👻", "🤡", "🐧", "🦚", "😄", "🚀"];
 
 export default function Memory() {
     const [boardData, setBoardData] = useState([]);
     const [flippedCards, setFlippedCards] = useState([]);
+    const [matchedCards, setMatchedCards] = useState([]);
 
     useEffect(() => {
+        initialize();
+    }, []);
+
+    const initialize = () => {
         const randomCards = [...board, ...board]
             .sort(() => Math.random() - 0.5)
-            .map((v) => v);
+            .map((mixed) => mixed);
         setBoardData(randomCards);
         setFlippedCards([]);
-    }, []);
-
-    const updateActiveCards = (i) => {
-        if (!flippedCards.includes(i)) {
-            if (flippedCards.length == 1) {
+        setMatchedCards([]);
+    }
 
-                setFlippedCards([...flippedCards, i]);
-            } else if (flippedCards.length == 2) {
-                setFlippedCards([i]);
+    const updateActiveCards = (iterate) => {
+        if (!flippedCards.includes(iterate)) {
+            if (flippedCards.length === 1) {
+                const firstIdx = flippedCards[0];
+                const secondIdx = iterate;
```

Comment:
```suggestion
                const firstIndex = flippedCards[0];
                const secondIndex = iterate;
```

this change should be applied to other iterations as well